### PR TITLE
Release chores

### PR DESCRIPTION
**Pull request by MaceWindu**
- update deps
- adjust some unstable tests/baselines
- remove net7 (EOL) support from cli and couple of T4 nugets
- fix latest oracle provider test issues
- update api baselines

### CODE REVIEW

**Review by sdanyliv — DISMISSED** [review]

File: Source/LinqToDB/DataProvider/Oracle/OracleProviderAdapter.cs [hunk 1]
```diff
@@ -1442,7 +1442,13 @@ public enum OracleDbType
...
 				Json         = 135,
 				ArrayAsJson  = 136,
-				ObjectAsJson = 137
+				ObjectAsJson = 137,
+
+				// Oracle 23
+				Vector         = 138,
+				Vector_Int8    = 139,
+				Vector_Float32 = 140,
+				Vector_Float64 = 141
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/Oracle/OracleProviderAdapter.cs:1451]:
```suggestion
				Vector_Float64 = 141,
```

**Review by viceroypenguin — APPROVED** [review]

**Review by igor-tkachev — APPROVED** [review]

File: Source/LinqToDB/DataProvider/Firebird/FirebirdSqlBuilder.cs [hunk 2]
```diff
@@ -172,7 +172,19 @@ protected override void BuildDataTypeFromDataType(DbDataType type, bool forCreat
...
+						StringBuilder.Append("CHAR CHARACTER SET OCTETS");
+					else
+						StringBuilder.Append(CultureInfo.InvariantCulture, $"CHAR({type.Length}) CHARACTER SET OCTETS");
+					break;
+				case DataType.VarBinary     :
+					if (type.Length == null || type.Length > 32_765)
+						StringBuilder.Append("BLOB");
+					else
+						StringBuilder.Append(CultureInfo.InvariantCulture, $"VARCHAR({type.Length}) CHARACTER SET OCTETS");
+					break;
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/Firebird/FirebirdSqlBuilder.cs:186]:
```suggestion
				case DataType.Binary when type.Length == null || type.Length < 1:
                    StringBuilder.Append("CHAR CHARACTER SET OCTETS");
                    break;

				case DataType.Binary:
                    StringBuilder.Append(CultureInfo.InvariantCulture, $"CHAR({type.Length}) CHARACTER SET OCTETS");
					break;

				case DataType.VarBinary when type.Length == null || type.Length > 32_765:
					StringBuilder.Append("BLOB");
                    break;

				case DataType.VarBinary:
					StringBuilder.Append(CultureInfo.InvariantCulture, $"VARCHAR({type.Length}) CHARACTER SET OCTETS");
					break;
```

File: Source/LinqToDB/DataProvider/Oracle/OracleProviderDetector.cs [hunk 3]
```diff
@@ -115,18 +119,25 @@ public override IDataProvider GetDataProvider(ConnectionOptions options, OracleP
...
+			// as connection string for DevArt has own (and actually more sane) format
+			// we cannot try to use incompatible provider
+			var canBeDevart = connectionString?.IndexOf("SERVER", StringComparison.OrdinalIgnoreCase) != -1;
+			var canBeOracle = connectionString?.IndexOf("DATA SOURCE", StringComparison.OrdinalIgnoreCase) != -1;
+
 			var fileName = typeof(OracleProviderDetector).Assembly.GetFileName();
 			var dirName  = Path.GetDirectoryName(fileName);
 
-			return File.Exists(Path.Combine(dirName ?? ".", OracleProviderAdapter.ManagedAssemblyName + ".dll"))
+			return canBeOracle && File.Exists(Path.Combine(dirName ?? ".", OracleProviderAdapter.ManagedAssemblyName + ".dll"))
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/Oracle/OracleProviderDetector.cs:132]: I'd rather see this as a bunch of `if()` statements:
```cs
if (canBeOracle && ...)
  return ..;
if (canBeDevart && ...)
  return ..;
...
```

File: Source/LinqToDB/DataProvider/Firebird/Firebird4SqlBuilder.cs [hunk 4]
```diff
@@ -32,5 +34,29 @@ protected override bool BuildJoinType(SqlJoinedTable join, SqlSearchCondition co
...
+						StringBuilder.Append("BINARY");
+					else
+						StringBuilder.Append(CultureInfo.InvariantCulture, $"BINARY({type.Length})");
+					break;
+				case DataType.VarBinary:
+					if (type.Length == null || type.Length > 32_765)
+						StringBuilder.Append("BLOB");
+					else
+						StringBuilder.Append(CultureInfo.InvariantCulture, $"VARBINARY({type.Length})");
+					break;
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/Firebird/Firebird4SqlBuilder.cs:54]:
```suggestion
				case DataType.Binary when type.Length == null || type.Length < 1):
					StringBuilder.Append("BINARY");
					break;
					
				case DataType.Binary:
					StringBuilder.Append(CultureInfo.InvariantCulture, $"BINARY({type.Length})");
					break;
					
				case DataType.VarBinary when type.Length == null || type.Length > 32_765):
					StringBuilder.Append("BLOB");
					break;
					
				case DataType.VarBinary:
					StringBuilder.Append(CultureInfo.InvariantCulture, $"VARBINARY({type.Length})");
					break;
```

**Review by viceroypenguin — DISMISSED** [review]

**Review by viceroypenguin — APPROVED** [review]

File: Source/LinqToDB/DataProvider/Firebird/Firebird4SqlBuilder.cs [hunk 5]
```diff
@@ -32,5 +34,32 @@ protected override bool BuildJoinType(SqlJoinedTable join, SqlSearchCondition co
 
 			return base.BuildJoinType(join, condition);
 		}
+
+		protected override void BuildDataTypeFromDataType(DbDataType type, bool forCreateTable, bool canBeNull)
+		{
+			switch (type.DataType)
+			{
+				case DataType.Guid  : StringBuilder.Append("BINARY(16)"); break;
+				case DataType.Binary when type.Length == null || type.Length < 1:
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/Firebird/Firebird4SqlBuilder.cs:43]:
```suggestion
				case DataType.Binary when type.Length is null or < 1:
```

File: Source/LinqToDB/DataProvider/Firebird/Firebird4SqlBuilder.cs [hunk 6]
```diff
@@ -32,5 +34,32 @@ protected override bool BuildJoinType(SqlJoinedTable join, SqlSearchCondition co
...
+					StringBuilder.Append("BINARY");
+					break;
+					
+				case DataType.Binary:
+					StringBuilder.Append(CultureInfo.InvariantCulture, $"BINARY({type.Length})");
+					break;
+					
+				case DataType.VarBinary when type.Length == null || type.Length > 32_765:
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/Firebird/Firebird4SqlBuilder.cs:51]:
```suggestion
				case DataType.VarBinary when type.Length is null or > 32_765:
```

File: Source/LinqToDB/DataProvider/Firebird/FirebirdSqlBuilder.cs [hunk 7]
```diff
@@ -172,7 +172,22 @@ protected override void BuildDataTypeFromDataType(DbDataType type, bool forCreat
 						base.BuildDataTypeFromDataType(type, forCreateTable, canBeNull);
 					break;
 
-				case DataType.VarBinary     : StringBuilder.Append("BLOB");                               break;
+				case DataType.Binary when type.Length == null || type.Length < 1:
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/Firebird/FirebirdSqlBuilder.cs:175]:
```suggestion
				case DataType.Binary when type.Length is null or < 1:
```

File: Source/LinqToDB/DataProvider/Firebird/FirebirdSqlBuilder.cs [hunk 8]
```diff
@@ -172,7 +172,22 @@ protected override void BuildDataTypeFromDataType(DbDataType type, bool forCreat
...
+                    StringBuilder.Append("CHAR CHARACTER SET OCTETS");
+                    break;
+
+				case DataType.Binary:
+                    StringBuilder.Append(CultureInfo.InvariantCulture, $"CHAR({type.Length}) CHARACTER SET OCTETS");
+					break;
+
+				case DataType.VarBinary when type.Length == null || type.Length > 32_765:
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/Firebird/FirebirdSqlBuilder.cs:183]:
```suggestion
				case DataType.VarBinary when type.Length is null or > 32_765:
```

**Review by viceroypenguin — APPROVED** [review]

**Pull request merged**